packages/@aws-cdk/aws-apigateway/lib/deployment.ts

```diff
@@ -121,20 +121,13 @@ class LatestDeploymentResource extends CfnDeployment {
    * add via `addToLogicalId`.
    */
   protected prepare() {
+    const stack = Stack.of(this);
+
     // if hash components were added to the deployment, we use them to calculate
     // a logical ID for the deployment resource.
     if (this.hashComponents.length > 0) {
       const md5 = crypto.createHash('md5');
-      this.hashComponents
-        .map(c => {
-          return Tokenization.resolve(c, {
-            scope: this,
-            resolver: new DefaultTokenResolver(new StringConcat()),
-            preparing: true,
-          });
-        })
-        .forEach(c => md5.update(JSON.stringify(c)));
-
+      this.hashComponents.map(x => stack.resolve(x)).forEach(c => md5.update(JSON.stringify(c)));
```

Comment:
You're reverting a part of this commit - https://github.com/aws/aws-cdk/commit/17fc967d. 
This specific piece of code was aimed at fixing https://github.com/aws/aws-cdk/issues/3705. Perhaps the fix wasn't accurate in using `StringConcat()`? 

I suspect (and hope so we don't regress) this will fail a unit test.

The original problem was you can't 'resolve' the entire stack before it was finished 'prepare'ing

Reply:
seems like #239 resolved the issue since cross stack reference resolution was moved from the element level to the stack level so we are all good.